tests/test_financial.py

```diff
@@ -90,13 +90,27 @@ def test_decimal_with_when(self):
 
 
 class TestPV:
-    def test_pv(self):
-        assert_almost_equal(npf.pv(0.07, 20, 12000, 0), -127128.17, 2)
-
-    def test_pv_decimal(self):
-        assert_equal(npf.pv(Decimal('0.07'), Decimal('20'), Decimal('12000'),
-                            Decimal('0')),
-                     Decimal('-127128.1709461939327295222005'))
+    @pytest.mark.parametrize("rate, periods, payments, future_value, exp_result", [
+                             (0.07, 20, 0, 12000, -3101.0280337664),
+                             (0.025, 4, 0, 1000, -905.9506447998),
+                             (-0.07, 56, 0, 1200000, -69845129.5182595000),
+                             (0.05, 20, 0, -12000, 4522.6737944760),
+                             (0.05, 356.5288568, 0, -25000.00, 0.0006971776),
+                             (0.05, 50, 0, 0, 0.0000000000),
+                             (0.05, 26, 500, 0, -7187.5926504975),
+                             ])
+    @pytest.mark.parametrize('number_type', [float, Decimal])
+    def test_pv(self, rate, periods, payments, future_value, exp_result, number_type):
+        result = npf.pv(rate, periods, payments, future_value)
+
+        if exp_result is not numpy.nan:
+            if number_type is Decimal:
+                assert result == pytest.approx(exp_result, rel=0.2)
```

Comment:
Please prefer the routines in [numpy.testing](https://numpy.org/doc/stable/reference/routines.testing.html) over those in pytest. 

In this case, we should not be testing that two values are _approximately_ equal, but that they are exactly equal as, when we are using Decimal, we should be keeping arbitrary precision.